packages/api/src/modules/data-viz/paymentFlat/paymentFlat.service.ts

```diff
@@ -0,0 +1,46 @@
+import { ObjectId } from "mongodb"; // Import the ObjectId type
+import paymentFlatPort from "../../../dataProviders/db/paymentFlat/paymentFlat.port";
+import dataBretagneService from "../../providers/dataBretagne/dataBretagne.service";
+import chorusService from "../../providers/chorus/chorus.service";
+import PaymentFlatAdapter from "./paymentFlatAdapter";
+
+export class PaymentFlatService {
+    private async getAllDataBretagneData() {
+        const ministries = await dataBretagneService.getMinistriesRecord();
+        const programs = await dataBretagneService.findProgramsRecord();
+        const domainesFonct = await dataBretagneService.getDomaineFonctRecord();
+        const refsProgrammation = await dataBretagneService.getRefProgrammationRecord();
+        return { programs, ministries, domainesFonct, refsProgrammation };
+    }
+
+    public async upDatePaymentsFlatCollection(lastChorusObjectId: ObjectId) {
+        const { programs, ministries, domainesFonct, refsProgrammation } = await this.getAllDataBretagneData();
+
+        const chorusCursor = await chorusService.chorusCursorFind({ _id: { $gt: lastChorusObjectId } });
+        let document = await chorusCursor.next();
+        let newChorusLastUpdate = lastChorusObjectId;
+
+        while (document != null) {
```

Comment:
On peut faire un do while je pense

Reply:
je ne suis pas sûre de comprendre le pourquoi ? je ne souhaite pas que le while soit exécuté si le document est null, et do while il l'exécute une première fois quand même non ?

Reply:
Oui tu peux faire le 1er cursor.next dans la boucle je crois

Reply:
mmhh.. mais si document est null toutes les autres choses que je fais dans la boucle petent non ?

Reply:
Merci Alice pour l'info! Est-ce qu'il y a un avantage particulier à utiliser le for plutôt que le while ? Dans la doc il presente les deux méthodes sans forcement spécifier qu'il faut prediliger une par rapport à l'autre. Et compte tenu que j'ai déjà fait tous les tests, je me demande si c'est vraiment pertinent de changer cette partie là ? Si oui, je me penche bien sûr dessus!